src/schema/models/schemaManager.ts

```diff
@@ -0,0 +1,45 @@
+import _ from 'lodash';
+import { inject, injectable } from 'tsyringe';
+import { Schemas, Schema, Tags } from './mapping';
+
+@injectable()
+export class SchemaManager {
+  public constructor(@inject(Schemas) private readonly schemas: Schemas) {}
+
+  public async getSchemas(): Promise<Schema[]> {
+    return this.schemas.getSchemas();
+  }
+
+  public async getSchema(name: string): Promise<Schema | undefined> {
+    return this.schemas.getSchema(name);
+  }
+
+  public async map(name: string, tags: Tags): Promise<Tags> {
+    const schema: Schema | undefined = await this.getSchema(name);
+    let tagsCopy = { ...tags };
+
+    if (schema) {
+      tagsCopy = this.mapTags(tags, schema);
+    }
+    return this.addSystemName(tagsCopy, name);
+  }
```

Comment:
is that the intended behavior? i think it should return an error to the user if the schema doesn't exist, not the same tags.

Reply:
implementation should be according to the spec, the spec says `404 Not Found`.

Reply:
Done